### PR TITLE
New pausable regions

### LINKED ISSUE #903: Add new pausable regions
Add new pausable regions: 
- priceDiscovery
- sequentialCommit

So if one of these regions is misfunctioning we don't have to stop all exchanges.

### PR DESCRIPTION
**Pull request by zajck**
Fix #903 

Adds two new pausable regions and changes the `RegionPaused()` error to include the region that reverted.

### CODE REVIEW

**Review by albertfolch-redeemeum — DISMISSED** [review]

File: test/protocol/PriceDiscoveryHandlerFacet.js [hunk 1]
```diff
@@ -504,7 +506,23 @@ describe("IPriceDiscoveryHandlerFacet", function () {
...
+
+            // Attempt to sequentially commit, expecting revert
+            await expect(
+              priceDiscoveryHandler
+                .connect(buyer)
+                .commitToPriceDiscoveryOffer(buyer.address, tokenId, priceDiscovery, { value: price })
+            )
+              .to.revertedWithCustomError(bosonErrors, RevertReasons.REGION_PAUSED)
+              .withArgs(PausableRegion.PriceDiscovery);
           });
```
Comment (levalleux-ludo) [inline test/protocol/PriceDiscoveryHandlerFacet.js:527]: Could be interesting to test that commit to a fixed price offer is still possible, even when the PriceDiscovery region is Paused

File: test/protocol/SequentialCommitHandlerTest.js [hunk 2]
```diff
@@ -592,7 +594,37 @@ describe("IBosonSequentialCommitHandler", function () {
...
+              // Attempt to sequentially commit, expecting revert
+              await expect(
+                sequentialCommitHandler
+                  .connect(buyer2)
+                  .sequentialCommitToOffer(buyer2.address, tokenId, priceDiscovery, { value: price2 })
+              )
+                .to.revertedWithCustomError(bosonErrors, RevertReasons.REGION_PAUSED)
+                .withArgs(PausableRegion.SequentialCommit);
+            });
+
```
Comment (levalleux-ludo) [inline test/protocol/SequentialCommitHandlerTest.js:615]: Could be interesting to test that commit to a fixed price offer and commit to a price discovery offer are still possible, even when the SequentialCommit region is Paused

**Review by levalleux-ludo — APPROVED** [review]

File: test/protocol/PriceDiscoveryHandlerFacet.js (hunk 1 above)
Comment (zajck) [reply test/protocol/PriceDiscoveryHandlerFacet.js:539]: Added (ouside "revert reasons" context).

File: test/protocol/SequentialCommitHandlerTest.js (hunk 2 above)
Comment (zajck) [reply test/protocol/SequentialCommitHandlerTest.js:615]: Added in appropriate files (exchangeHandlerTests and priceDiscoveryTest)

**Review by levalleux-ludo — APPROVED** [review]

**Review by albertfolch-redeemeum — APPROVED** [review]

**Pull request merged**